src/pulselistener/pulselistener/listener.py

```diff
@@ -111,99 +116,89 @@ def list_differential(self):
 
     async def build_consumer(self, *args, **kwargs):
         '''
-        Query phabricator differentials regularly
+        Main consumer:
+        * Query phabricator differentials regularly
+        * Check queued builds
         '''
-        if self.mode != MODE_PHABRICATOR_POLLING:
-            return
-
         while True:
 
             # Get new differential ids
-            for diff in self.list_differential():
-                try:
-                    # Load revision to get repository
-                    rev = self.api.load_revision(diff['revisionPHID'])
+            if self.mode == MODE_PHABRICATOR_POLLING:
+                await self.poll_phabricator()
 
-                    logger.info('Triggering task from polling', diff=diff['id'])
-                    await self.trigger_task(diff, rev['fields']['repositoryPHID'])
-                except Exception as e:
-                    logger.error('Failed to trigger task from polling', error=str(e))
+                # Sleep a bit before trying new diffs
+                await asyncio.sleep(60)
 
-            # Sleep a bit before trying new diffs
-            await asyncio.sleep(60)
+            elif self.mode == MODE_PHABRICATOR_WEBHOOK:
+                await self.run_builds()
 
-    async def new_code_review(self, request):
+                # Sleep just a bit between two runs
+                await asyncio.sleep(2)
```

Comment:
Minor: To avoid the sleep, you could use a queue where you put elements when you receive the HTTP request from HarborMaster, and here you wait for a new element in the queue and call self.run_builds().